### DISCUSSION TITLE
Make renovate only do what's possible

### DISCUSSION THREAD

**Discussion by jankal** [renovatebot/renovate#28302] [Request Help]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Mend Renovate hosted app on github.com

### Please tell us more about your question or problem

I am using renovate with Shopware - a Symfony based e-commerce framework. Shopware is written in PHP and thus uses Composer as its package manager. Shopware has the concept of plugins which can be installed through composer.

Those plugins have version constraints on the shopware frameworks core packages like `shopware/core` or `shopware/storefront`. If there is a bugfix for a shopware plugin, usually a new fix-version is released (before `2.1.0` -> `2.1.1`).
It happens that one plugin supports multiple shopware core versions (e.g. `6.5` and `6.6` - explanation of the versioning scheme in https://github.com/renovatebot/renovate/discussions/28294) and thus in case a fix for both versions is released, those fixes are often backported and an additional `1.4.0` -> `1.4.1` update is released to enable users of shopware core packages compatible with `1.x` of the plugin to receive the fix too.

Now if I have the following set of packages in my `composer.json`:

```json
{
    "require": {
        "php": ">=8.3.0",
        "composer-runtime-api": "^2.0",
        "shopware/core": "6.5.8.8",
        "shopware/elasticsearch": "6.5.8.8",
        "shopware/storefront": "6.5.8.8",
        "symfony/amqp-messenger": "^7.0",
        "symfony/flex": "~2",
        "symfony/runtime": "^5.0|^6.0",
        "vendor/plugin": "^6.2"
    }
}
```

And `vendor/plugin` has the following versions available:
- `6.2.0` -> compatible with `"shopware/core": "~6.5"`
- `6.2.1` as a bugfix -> compatible with `"shopware/core": "~6.5"`
- `7.0.0`-> compatible with `"shopware/core": "~6.6"`
- `7.0.1` as a bugfix-> compatible with `"shopware/core": "~6.6"`

And my currently installed (thus locked) version is `"vendor/plugin": "6.2.0"`.

Renovate will create a PR to update `vendor/plugin` to `7.0.1` while my project still uses `"shopware/core": "6.5.8.8"`.
Sure enough there will be another PR for upgrading `shopware/core` to the latest `6.6.x` version.

BUT both PRs will not be mergeable as they are interdependent on each other. But PRs will have a **Artifact update problem** as `composer update` when tasked with only updating one of them to the highest possible version will inevitably fail and thus a `composer.lock` cannot be generated.

**The modifications in `composer.json` done by renovate will result in an uninstallable set of packages as dependencies of the updated packages are not respected when suggesting the update.** - And multiple interdependent updates cannot be combined.

How can I configure renovate to only update some packages (e.g. using a `packageRule`) to the highest possible versions (inter-dependency wise) while allowing other packages to be updated even further?

**Comment by rarkins** [comment]
If two packages must be updated together then you need to add a grouping rule to group them together. If you think that all users would benefit from such a grouping then it can be added to the group:recommended preset to save it needing to be configured by every user.

Whether Renovate could one day automatically detect such cross-dependencies is another matter. Unless the ecosystem supports that concept in metadata then it may not be possible for automatic grouping.
